### DISCUSSION TITLE
BCs for a fixed-pressure 2D RZ channel flow

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#27574] [Q&A Modules: Navier-Stokes] [answered]
Hello MOOSE experts:

I'm learning Navier Stokes, and try to simulate a 2D symmetric channel flow which describe as follow:

![simulation_settings](https://github.com/idaholab/moose/assets/111510550/2c8e596f-c3fd-4db8-84b3-e8b312eb0abf)

Here's my input block using **NavierStokesFV Action** 
```
[Modules]
  [NavierStokesFV]
    compressibility = 'weakly-compressible'
    add_energy_equation = true

    density = 'rho'
    dynamic_viscosity = 'mu'
    thermal_conductivity = 'k'
    specific_heat = 'cp'

    initial_temperature = 800
    initial_pressure = 900

    inlet_boundaries = 'right'
    momentum_inlet_types = 'fixed-pressure'
    momentum_inlet_function = 'boundary-pressure'
    energy_inlet_types = 'fixed-temperature'
    energy_inlet_function = 'boundary-temperature'

    wall_boundaries = 'top bottom left'
    momentum_wall_types = 'noslip noslip symmetry'
    energy_wall_types = 'heatflux heatflux heatflux'
    energy_wall_function = '0 0 0'
  []
[]

[Functions]
  [boundary-pressure]
    type = ParsedFunction
    expression = '900 + 0.02 * y'
  []
  [boundary-temperature]
    type = ParsedFunction
    expression = '800'
  []
[]
```

With complete input file
[2d-rz-wcns.txt](https://github.com/idaholab/moose/files/15244304/2d-rz-wcns.txt)

The run converge well but I'm not sure if the boundary conditions are set correctly, because the results seems strange (as shown in left side of following picture, non-zero axial velocity on right boundary), and differs with result by COMSOL (shown in right, which seems more reasonable for me)

<img width="311" alt="vel_comp" src="https://github.com/idaholab/moose/assets/111510550/c253b40e-a6d9-4e4f-9a27-44c97769e8d8">

Could someone help to check my input? 

Thanks in advance for help!

**Comment by GiudGiud** [comment]
Hello

Why are you expecing non-zero velocity on the right boundary? It's a fixed pressure BC not a no-slip wall for velocity. 
The fixed-pressure BC in MOOSE is coded for a fully-developed outlet boundary condition.

Can you draw streamlines in both codes? I wonder where the flow is going in the COMSOL simulation. It seems to flow down, but never up after, and it cant go through the symmetry BC so where does it go? Must be right

Guillaume

**Reply by WayneX-Wong** [reply]
Hello Guillaume

I think for inlet on the right boundary, there should be only normal velocity and zero axial velocity. I don't know if I'm getting it right

Following is the streamline of both framework (MOOSE on the left-side)
<img width="321" alt="streamline" src="https://github.com/idaholab/moose/assets/111510550/f6f655c7-7f7e-487b-a127-8d15b4f1b4e1">

And following is a comparison of the velocity components (MOOSE on the left-side)
<img width="579" alt="compon" src="https://github.com/idaholab/moose/assets/111510550/f98bee0a-4815-4081-a2c5-f2d3fbda7615">

Both results show that the working fluid flows from the upper-right high pressure region to the lower-right low pressure region (Pressure decreases linearly from top to bottom on the right boundary)

Difference seems to be that COMSOL specifies a normal flow direction at the pressure inlet while MOOSE does not

**Reply by GiudGiud** [reply, marked as answer]
> I think for inlet on the right boundary, there should be only normal velocity and zero axial velocity. I don't know if I'm getting it right

We don't impose this constraint on velocity in the fixed pressure BC.
You could add a dirichletBC on velocity_z to impose the 0 tangential velocity

**Reply by WayneX-Wong** [reply]
Thanks for your suggestion, now the results are same. I didn't realize that I could add another BCs block outside Module block.

I still wonder if it's necessary to constrain a normal flux on a pressure BCs... Seems Commercial software like Fluent and Comsol have this by default, or artificially specify another flow direction. Possibly they do this for improving convergence.

**Reply by GiudGiud** [reply]
maybe @grmnptr will know
